guilford_dps_offenders.yaml

```diff
@@ -0,0 +1,330 @@
+          transformsIfTrue:
+          - !<transforms.ValueTransform>
+            value: null
+          transformsIfFalse:
+          - !<transforms.ColumnTransform>
+            column: "OFFENDER PRIMARY LANGUAGE CODE"
+      ol.primary:
+        type: "ol.primary"
+        transforms:
+        - !<transforms.ValueTransform>
+          value: "Yes"
+        - !<transforms.ParseBoolTransform> {}
+    conditions:
+    - !<conditions.BooleanRegexCondition>
+      column: "OFFENDER PRIMARY LANGUAGE CODE"
+      pattern: "(?i)(\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?)"
+      reverse: true
+    name: "offendercommunication"
+
+  offenderlocation:
+    fqn: "ol.location"
+    entitySetName: "GuilfordBirthPlaces"
+    propertyDefinitions:
+      general.id:
+        type: "general.id"
+        column: "OFFENDER NC DOC ID NUMBER"
+      location.city:
+        type: "location.city"
+        transforms:
+        - !<transforms.BooleanRegexTransform>
+          column: "CITY WHERE OFFENDER BORN"
+          pattern: "(\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?)"
+          transformsIfTrue:
+          - !<transforms.ValueTransform>
+            value: null
+          transformsIfFalse:
+          - !<transforms.ColumnTransform>
+            column: "CITY WHERE OFFENDER BORN"
+      ol.county:
+        type: "ol.county"
+        transforms:
+        - !<transforms.BooleanRegexTransform>
+          column: "NC COUNTY WHERE OFFENDER BORN"
+          pattern: "(\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?)"
+          transformsIfTrue:
+          - !<transforms.ValueTransform>
+            value: null
+          transformsIfFalse:
+          - !<transforms.ColumnTransform>
+            column: "NC COUNTY WHERE OFFENDER BORN"
+      location.state:
+        type: "location.state"
+        transforms:
+        - !<transforms.BooleanRegexTransform>
+          column: "STATE WHERE OFFENDER BORN"
+          pattern: "(\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?)"
+          transformsIfTrue:
+          - !<transforms.ValueTransform>
+            value: null
+          transformsIfFalse:
+          - !<transforms.ColumnTransform>
+            column: "STATE WHERE OFFENDER BORN"
+      ol.country:
+        type: "ol.country"
+        transforms:
+        - !<transforms.BooleanRegexTransform>
+          column: "COUNTRY WHERE OFFENDER BORN"
+          pattern: "(\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?\\?)"
+          transformsIfTrue:
+          - !<transforms.ValueTransform>
+            value: null
+          transformsIfFalse:
+          - !<transforms.ColumnTransform>
+            column: "COUNTRY WHERE OFFENDER BORN"
+    name: "offenderlocation"
```

Comment:
can you also add a condition on this entity set to only read into it if at least one column in it is not null?

Reply:
done